janusgraph-core/src/test/java/org/janusgraph/diskstorage/util/CompletableFutureUtilTest.java

```diff
@@ -124,13 +124,11 @@ public void shouldFailMapUnwrapWithSuppressedExceptions() {
             otherFuture.completeExceptionally(new IllegalArgumentException());
             futureMap.put(new Object(), otherFuture);
         }
-        ExecutionException executionException = Assertions.assertThrows(ExecutionException.class, () -> unwrap(futureMap));
-        Assertions.assertTrue(executionException.getCause() instanceof IllegalStateException);
-        Throwable[] suppressedExceptions = executionException.getSuppressed();
+        IllegalStateException exception = Assertions.assertThrows(IllegalStateException.class, () -> unwrap(futureMap));
+        Throwable[] suppressedExceptions = exception.getSuppressed();
         Assertions.assertEquals(10, suppressedExceptions.length);
         for(int i=0; i<10; i++){
-            Assertions.assertTrue(suppressedExceptions[i] instanceof ExecutionException &&
-                suppressedExceptions[i].getCause() instanceof IllegalArgumentException);
+            Assertions.assertTrue(suppressedExceptions[i] instanceof IllegalArgumentException);
         }
```

Comment:
📝 Based on the code of `CompletableFutureUtil#unwrap` and `CompletableFutureUtil#unwrapExecutionException`, it would make sense to me that the exceptions returned are not `ExecutionException`s but their "unwrapped" counterpart (`IllegalStateException` or `IllegalArgumentException`).

Reply:
You are right. Sorry I forgot to change that previously. Initially it was returning ExecutionException but after some fixed I changed this behavior and completely forgot about these Unit tests.